### PR TITLE
ref(related_issues): TraceTimelineOrRelatedIssue component

### PR DESCRIPTION
**Pull request by armenzg**
This helps test the TraceLink, TraceTimeline and TraceIssueEvent as a collective of related components.

Before, we could not test the behaviour of the TraceLink.

### CODE REVIEW

File: static/app/views/issueDetails/groupEventHeader.tsx [hunk 1]
```diff
@@ -8,9 +8,8 @@ import type {Group} from 'sentry/types/group';
 import type {Project} from 'sentry/types/project';
 import useOrganization from 'sentry/utils/useOrganization';
 import {GroupEventCarousel} from 'sentry/views/issueDetails/groupEventCarousel';
-import {TraceLink} from 'sentry/views/issueDetails/traceTimeline/traceLink';
-import {TraceTimeline} from 'sentry/views/issueDetails/traceTimeline/traceTimeline';
```
Comment (armenzg) [inline static/app/views/issueDetails/groupEventHeader.tsx:12]: We are moving the rendering of `TraceLink` and `TraceTimeline` within the same new component `TraceTimeLineOrRelatedIssue`.

File: static/app/views/issueDetails/traceTimeline/traceTimeline.tsx [hunk 2]
```diff
@@ -34,12 +33,8 @@ export function TraceTimeline({event}: TraceTimelineProps) {
     timelineStatus = 'no_trace_id';
   }
 
-  const showTraceRelatedIssue =
-    timelineStatus !== 'shown' && oneOtherIssueEvent !== undefined;
-
   useRouteAnalyticsParams({
     trace_timeline_status: timelineStatus,
-    has_related_trace_issue: showTraceRelatedIssue,
```
Comment (armenzg) [inline static/app/views/issueDetails/traceTimeline/traceTimeline.tsx:42]: Route analytics for the trace-related issue are moved within the new component `TraceTimeLineOrRelatedIssue`.

File: static/app/views/issueDetails/traceTimeline/traceTimeline.tsx [hunk 3]
```diff
@@ -81,7 +76,6 @@ export function TraceTimeline({event}: TraceTimelineProps) {
           </TimelineWrapper>
         </Fragment>
       )}
-      {showTraceRelatedIssue && <TraceIssueEvent event={oneOtherIssueEvent} />}
```
Comment (armenzg) [inline static/app/views/issueDetails/traceTimeline/traceTimeline.tsx:84]: This change decouples the rendering of a related issue from the trace timeline.

File: static/app/views/issueDetails/traceTimelineOrRelatedIssue.spec.tsx [hunk 4]
```diff
@@ -100,13 +101,14 @@ describe('TraceTimeline', () => {
...
-    render(<TraceTimeline event={event} />, {organization});
+    render(<TraceTimeLineOrRelatedIssue event={event} />, {
+      organization,
+    });
     expect(await screen.findByLabelText('Current Event')).toBeInTheDocument();
 
     await userEvent.hover(screen.getByTestId('trace-timeline-tooltip-1'));
     expect(await screen.findByText('You are here')).toBeInTheDocument();
     expect(useRouteAnalyticsParams).toHaveBeenCalledWith({
-      has_related_trace_issue: false,
```
Comment (armenzg) [inline static/app/views/issueDetails/traceTimelineOrRelatedIssue.spec.tsx:109]: We do not use the `false` value in Ampltitude, thus, this is fine.

File: static/app/views/issueDetails/groupEventHeader.tsx [hunk 5]
```diff
@@ -20,23 +19,11 @@ type GroupEventHeaderProps = {
...
-      {!isLoading && (
-        <StyledTraceLink>
-          {/* This message is used when showing a trace-related issue */}
-          {oneOtherIssueEvent && <span>One other issue appears in the same trace.</span>}
-          <TraceLink event={event} />
-        </StyledTraceLink>
-      )}
-      {/* This renders either the trace timeline or the trace-related issue */}
-      <TraceTimeline event={event} />
+      <TraceTimeLineOrRelatedIssue event={event} />
```
Comment (armenzg) [inline static/app/views/issueDetails/groupEventHeader.tsx:26]: A single container to render all three.

File: static/app/views/issueDetails/traceTimelineOrRelatedIssue.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    params = {
+      has_related_trace_issue: true,
+    };
+  }
+  useRouteAnalyticsParams(params);
+
+  if (isLoading) {
+    return null;
+  }
+  return (
```
Comment (armenzg) [inline static/app/views/issueDetails/traceTimelineOrRelatedIssue.tsx:29]: This block is a modified version from groupEventHeader:
https://github.com/getsentry/sentry/blob/9c87b800ff4776af661dc2b0b270a32e90d5c8e5/static/app/views/issueDetails/groupEventHeader.tsx#L31-L39

File: static/app/views/issueDetails/traceTimelineOrRelatedIssue.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  const {isLoading, oneOtherIssueEvent} = useTraceTimelineEvents({
+    event,
+  });
+  let params = {};
+  if (!isLoading && oneOtherIssueEvent !== undefined) {
+    params = {
+      has_related_trace_issue: true,
+    };
+  }
+  useRouteAnalyticsParams(params);
```
Comment (armenzg) [inline static/app/views/issueDetails/traceTimelineOrRelatedIssue.tsx:24]: Partially moved the analytic reporting here from `TraceTimeline`.

File: static/app/views/issueDetails/traceTimelineOrRelatedIssue.spec.tsx [hunk 8]
```diff
@@ -205,12 +203,13 @@ describe('TraceTimeline', () => {
...
 
-    render(<TraceTimeline event={event} />, {
-      organization,
-    });
+    render(<TraceTimeLineOrRelatedIssue event={event} />, {organization});
 
     // Instead of a timeline, we should see the other related issue
     expect(await screen.findByText('Slow DB Query')).toBeInTheDocument();
+    expect(
+      await screen.findByText('One other issue appears in the same trace.')
```
Comment (armenzg) [inline static/app/views/issueDetails/traceTimelineOrRelatedIssue.spec.tsx:211]: This is the test we could not add before.

File: static/app/views/issueDetails/traceTimelineOrRelatedIssue.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+import {TraceLink} from './traceTimeline/traceLink';
+import {TraceTimeline} from './traceTimeline/traceTimeline';
+import {useTraceTimelineEvents} from './traceTimeline/useTraceTimelineEvents';
+
+export function TraceTimeLineOrRelatedIssue({event}: {event: Event}) {
+  // This is also called within the TraceTimeline component but caching will save a second call
+  const {isLoading, oneOtherIssueEvent} = useTraceTimelineEvents({
+    event,
+  });
+  let params = {};
```
Comment (scttcper) [inline static/app/views/issueDetails/traceTimelineOrRelatedIssue.tsx:18]: right now our settings allow you to set any property but in the future the type will be `{}` which will prevent this
```suggestion
  let params: Record<string, boolean> = {};
```

**Review by scttcper — APPROVED** [review]

**Pull request merged**